storage/statedb/sync.go

```diff
@@ -117,9 +122,14 @@ func (s *TrieSync) AddSubTrie(root common.Hash, depth int, parent common.Hash, c
 		return
 	}
 	if s.bloom.Contains(root[:]) {
-		key := root.Bytes()
-		blob, _ := s.database.ReadStateTrieNode(key)
-		if local, err := decodeNode(key, blob); local != nil && err == nil {
+		// Bloom filter says this might be a duplicate, double check
+		if s.exist != nil {
+			if _, ok := s.exist.Get(root); ok {
```

Comment:
Don't we need to check `s.database.HasStateTrieNode(root[:])` if `ok` is `false`? 
As `exist` is a cache, not a map.

Reply:
Right. The issue was the DB has a trie node which is not written by migration.
Therefore, this PR prevent to read DB. I will write more descriptions in this PR and explain the issue in offline.

Reply:
Thanks for the explanation! How about leaving a comment that fast sync is the only case that checks the database?

Reply:
Thanks. I will add a comment. :)